org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/binding/BindingsView.java

```diff
@@ -252,7 +260,42 @@ public void connectionRemoved(IConnectedEngineFacade server) {
     for (var server : SonarLintCorePlugin.getServersManager().getServers()) {
       server.addConnectedEngineListener(serverListener);
     }
-
+    
+    // add listener for when project is opened / closed / deleted
+    projectListener = event -> {
+      if (event.getBuildKind() == IResourceChangeEvent.PRE_CLOSE
+        || event.getBuildKind() == IResourceChangeEvent.PRE_DELETE) {
+        refreshView();
```

Comment:
Won't it fire every time a resource is deleted, like a file or a folder?

Reply:
No the event is only for projects